### PR TITLE
[cases] CaseScreen bugs

### PR DESCRIPTION
**Pull request by arfamomin**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Change eligibility requirements to be darkGray and change to black when clicked 
- Added updated warning icon 
- Made entire StatusUpdatesBar clickable, cleaned up divs here too  
- Also in StatusUpdatesBar, fixed alignment of the text
- Fixed? date issue being one day behind (need to check with Retool stuff)
- Added check in getStatusColor function that checks if string passed in is a valid status via StatusOptions const (array of all the colored status options enums); updated function keeps us from comparing strings directly 

### Screenshots
[//]: # 'Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder.'

This is me adding 'Action Required' temporarily into the GreenStatusOptions enum and it updates accordingly. 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Order of files changed works, main stuff is in utils and types file

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Need to make sure date conversion works when realtime dates are added via Retool and that whole configuration. 

### Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ronniebeggs

### CODE REVIEW

File: .idea/vcs.xml
Comment (ronniebeggs) [inline .idea/vcs.xml:1]: Is this from IntelliJ? I don't think we should merge this.

File: src/Components/PressableRequirement/styles.ts
Comment (ronniebeggs) [inline src/Components/PressableRequirement/styles.ts:1]: Looks better! Can we change the `requirementContainer` style so it has `paddingVertical: 20`. Also add `justifyContent: 'center'` to the `innerRequirementBox`. That way single lines of text are centered properly within the component.

File: src/Components/StatusUpdatesBar/StatusUpdatesBar.tsx [hunk 1]
```diff
@@ -19,22 +19,26 @@ export default function StatusUpdatesBar({
...
+        <View style={styles.topContainer}>
+          <Text style={styles.statusText}>Case Status:</Text>
+
           <Text style={styles.buttonText}>View all updates</Text>
           <RightCaret />
-        </TouchableOpacity>
-      </View>
-      <View style={[styles.statusContainer, statusColor.background]}>
-        <Text style={[styles.statusTextColor, statusColor.text]}>{status}</Text>
+        </View>
```
Comment (ronniebeggs) [inline src/Components/StatusUpdatesBar/StatusUpdatesBar.tsx]: In the process of moving the button, the "View all updates" text was spaced further away from the RightCaret icon. Can we create a div that holds these items so they don't get spread apart?

File: src/Components/StatusUpdatesBar/StatusUpdatesBar.tsx [hunk 2]
```diff
@@ -19,22 +19,26 @@ export default function StatusUpdatesBar({
...
-      <View style={styles.topContainer}>
-        <Text style={styles.statusText}>Case Status:</Text>
-        <TouchableOpacity
-          style={styles.updatesButton}
-          onPress={() => {
-            router.push({ pathname: `AllCases/Updates/${caseUid}` });
-          }}
-        >
+    <TouchableOpacity
+      style={styles.updatesButton}
```
Comment (ronniebeggs) [inline src/Components/StatusUpdatesBar/StatusUpdatesBar.tsx]: I think it'd be cleaner to make this full container a button by switching the View and TouchableOpacity components.  That way the outer container becomes a button with the `container` styling, while the previous TouchableOpacity becomes a view with the `updatesButton` styling. Make sure to change the names though!

File: src/app/(BottomTabNavigation)/AllCases/utils.ts [hunk 3]
```diff
@@ -36,8 +36,23 @@ export async function fetchAllCases(userUid: UserUid): Promise<Case[]> {
...
+    'Jun',
+    'Jul',
+    'Aug',
+    'Sep',
+    'Oct',
+    'Nov',
+    'Dec',
+  ];
+  const monthName = months[date.getUTCMonth()];
+  return `${monthName} ${date.getUTCDate()}, ${date.getUTCFullYear()}`;
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/utils.ts]: I'm not sure if this will fix the issue. I think this is a deeper issue than we think because we also have to consider different timezones, daylight savings etc. Look into this [article here](https://javascript.plainenglish.io/react-native-getting-user-device-timezone-and-converting-utc-time-stamps-with-offset-697902d7803) and see what you can find. Assume all dates in our database are stored in UTC time, and we must convert them to the user's specific timezone.

File: src/app/(BottomTabNavigation)/AllCases/utils.ts [hunk 4]
```diff
@@ -36,8 +36,23 @@ export async function fetchAllCases(userUid: UserUid): Promise<Case[]> {
...
 }
 
 export function getStatusColor(status: string) {
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/AllCases/utils.ts:55]: I have an idea for the `getStatusColor` function. We should create types for each status color type, such as `GreenStatusOptions`, and set it equal to the union of all options: `'In Progress' | 'New Case' | ...`. Do so for each of the four color options. Then union these options to create `CaseStatusOptions` type. That way if we want to adjust status options, we only have to do it in one place.

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
Everything's looking good so far! Just a few comments to address. I don't know how to mitigate the button feedback while scrolling but I'll ask david in the meantime.

File: .idea/vcs.xml
Comment (arfamomin) [reply .idea/vcs.xml:1]: oops that was an accident will delete in next commit

File: src/Components/PressableRequirement/styles.ts
Comment (arfamomin) [reply src/Components/PressableRequirement/styles.ts:1]: 🫡

File: src/Components/StatusUpdatesBar/StatusUpdatesBar.tsx (hunk 1 above)
Comment (arfamomin) [reply src/Components/StatusUpdatesBar/StatusUpdatesBar.tsx]: fixed

File: src/Components/StatusUpdatesBar/StatusUpdatesBar.tsx (hunk 2 above)
Comment (arfamomin) [reply src/Components/StatusUpdatesBar/StatusUpdatesBar.tsx]: I ended up just removing the view div so all the styling for the button is in updatesButton and then textContainer and statusContainer define the styling for the content on the button.

File: src/app/(BottomTabNavigation)/AllCases/utils.ts (hunk 3 above)
Comment (arfamomin) [reply src/app/(BottomTabNavigation)/AllCases/utils.ts]: I have the code using the moment stuff from the article already so if I should change from the built in Intl. DateTimeFormat object, I can!

File: src/app/(BottomTabNavigation)/AllCases/utils.ts [hunk 5]
```diff
@@ -36,18 +43,39 @@ export async function fetchAllCases(userUid: UserUid): Promise<Case[]> {
...
   if (
-    status === 'In Progress' ||
-    status === 'New Case' ||
-    status === 'Settled' ||
-    status === 'Appeal' ||
-    status === 'Payment Processing' ||
-    status === 'Payment Distributed'
+    Object.values(GreenStatusOptions).includes(
+      status as unknown as GreenStatusOptions,
+    )
```
Comment (philipye314) [inline src/app/(BottomTabNavigation)/AllCases/utils.ts]: hello arfa,

can you possibly remove the `as unknown as GreenStatusOptions` and subsequent occurences? It just seems a little bit redundant. functionally it won't do anything different I think

Warm regards,
Philip

File: src/Components/PressableRequirement/styles.ts [hunk 6]
```diff
@@ -8,10 +8,11 @@ export default StyleSheet.create({
     paddingTop: 20,
     columnGap: 20,
     paddingHorizontal: '6%',
```
Comment (philipye314) [inline src/Components/PressableRequirement/styles.ts]: paddingTop is overwritten by paddingVertical, so you can remove this! Also, for consistency sake, change `paddingHorizontal: '6%'` to `paddingHorizontal: 25`

_Sent from Smart Toilet_
Best,
Philip

**Review by philipye314 — COMMENTED** [review]
meow

File: src/app/(BottomTabNavigation)/AllCases/utils.ts (hunk 5 above)
Comment (arfamomin) [reply src/app/(BottomTabNavigation)/AllCases/utils.ts]: done

File: src/types/types.tsx [hunk 7]
```diff
@@ -88,3 +88,26 @@ export interface Update {
...
+}
+
+export enum RedStatusOptions {
+  'Action Required',
+}
+
+export const StatusOptions: string[] = [
+  ...(Object.values(GreenStatusOptions) as string[]),
+  ...(Object.values(YellowStatusOptions) as string[]),
+  ...(Object.values(RedStatusOptions) as string[]),
```
Comment (ronniebeggs) [inline src/types/types.tsx]: For the sake of simplifying this (since I honestly don't know what's happening) let's assume that for any status input that we don't recognize, we'll return a default grey color. That way we can avoid introducing new errors into the codebase.

So, let's get rid of this `StatusOptions` type and change the util function accordingly.

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
Looks good! I tried standardizing shadow styling to apply across the app. Feel free to check how I did that. Please just fix the util function and it should be ready to merge!

File: src/types/types.tsx (hunk 7 above)
Comment (arfamomin) [reply src/types/types.tsx]: I removed the `StatusOptions` type and deleted the extra code in at the top of the function that dealt with the error.  Let me know if that isn't what you meant and I can change.

**Comment by arfamomin** [comment]
Looked over how you standardized the shadow styling and double checked if there was anywhere else to apply it -- saw the update view when you click into a specific update had similar styling except the border radius was 15. I applied it to that in my latest commit but if it's bad practice to override one property, I can revert back to the previous commit.

**Comment by ronniebeggs** [comment]
> Looked over how you standardized the shadow styling and double checked if there was anywhere else to apply it -- saw the update view when you click into a specific update had similar styling except the border radius was 15. I applied it to that in my latest commit but if it's bad practice to override one property, I can revert back to the previous commit.

Great catch! I don't think it's necessarily bad practice to override one property, it just comes down to preference. If we consider the alternative, we could remove the `borderRadius` prop from the global `shadowBorder` style, as border behavior differs slightly depending on which component it's applied to. But then perhaps in the future, if we decided to change the styling of these boxes, we'd have to think about how to change every component separately. TLDR it's probably fine how it is.
